lxd/instance/drivers/driver_common.go

```diff
@@ -1106,6 +1117,26 @@ func (d *common) restoreCommon(inst instance.Instance, source instance.Instance)
 		return false, nil, fmt.Errorf("Failed restoring snapshot rootfs: %w", err)
 	}
 
+	// Restore attached volume snapshots.
+	if len(restoreVolumes) > 0 {
+		storageCache := storagePools.NewStorageCache(pool) // Create storage cache for pool lookups.
+		for _, volume := range restoreVolumes {
+			volName, snapName, _ := api.GetParentAndSnapshotName(volume.Name)
+
+			d.logger.Debug("Restoring attached volume snapshot", logger.Ctx{"pool": volume.Pool, "volume": volName, "snapshot": snapName, "project": volume.Project})
+
+			pool, err := storageCache.GetPool(volume.Pool)
+			if err != nil {
+				return false, nil, fmt.Errorf("Failed loading storage pool %q: %w", volume.Pool, err)
+			}
+
+			err = pool.RestoreCustomVolume(volume.Project, volName, snapName, d.op)
+			if err != nil {
+				return false, nil, fmt.Errorf("Failed restoring volume %q snapshot %q: %w", volume.Name, snapName, err)
```

Comment:
this error should include the pool name as it may be different than the instance's pool name